front/src/common/Map/Settings/MapSettingsSpeedLimits.tsx

```diff
@@ -39,31 +56,28 @@ function FormatSwitch(props) {
     );
   };
 
-  const getTagsList = async (_zoom, _params) => {
-    try {
-      const tagsList = await get(`/editoast/infra/${infraID}/speed_limit_tags/`);
+  const changeTagsListToObject = (tagsListArray: string[]) => {
+    const tagsListObject = [{ key: t('noSpeedLimitByTag'), value: t('noSpeedLimitByTag') }].concat(
+      tagsListArray.slice().map((tag) => ({ key: tag, value: tag }))
+    );
+    setSpeedLimitsTags(tagsListObject);
+  };
 
-      // construct an object from array to add "undefined" value
-      const tagsListObject = [{ key: 'undefined', value: t('noSpeedLimitByTag') }].concat(
-        tagsList.map((tag) => ({ key: tag, value: tag }))
-      );
+  useEffect(() => {
+    setSpeedLimitsTags(undefined);
+    if (tagsList) changeTagsListToObject(tagsList);
+  }, [infraID, tagsList]);
 
-      setSpeedLimitsTags(tagsListObject);
-    } catch (e) {
+  useEffect(() => {
+    if (isGetSpeedLimitTagsError) {
```

Comment:
You could use error like that : 
```suggestion
    if (isGetSpeedLimitTagsError && error) {
      dispatch(
          setFailure({
            name: t('errorMessages.unableToRetrieveTags'),
            message: `${(error as ApiError)?.data?.message || (error as SerializedError)?.message}`,
```

Reply:
Good comment thank you.